lib/charms/mongodb/v1/mongodb_backups.py

```diff
@@ -806,3 +819,35 @@ def get_backup_error_status(self, backup_id: str) -> str:
                 return backup.get("error", "")
 
         return ""
+
+    def are_s3_configurations_provided(self) -> bool:
+        """Returns True if not all necessary configurations for s3 are provided.
+
+        Minimum necessary s3 credentials can be found by reading the PBM page:
+        https://docs.percona.com/percona-backup-mongodb/reference/configuration-options.html#s3-type-storage-options
+        """
+        if not self.model.get_relation(S3_RELATION):
+            logger.info("No configurations for backups, not relation to s3-charm.")
+            return False
+
+        provided_configs = self._get_pbm_configs()
+        if (
+            "storage.s3.credentials.access-key-id" not in provided_configs
+            or "storage.s3.credentials.secret-access-key" not in provided_configs
```

Comment:
Safer with the following as we're also testing on the value 
```suggestion
            not provided_configs.get("storage.s3.credentials.access-key-id")
            or not provided_configs.get("storage.s3.credentials.secret-access-key")
```